rp2040-hal-examples/Cargo.toml

```diff
@@ -0,0 +1,36 @@
+[package]
+name = "rp2040-hal-examples"
+version = "0.1.0"
+edition = "2021"
+authors = ["The rp-rs Developers"]
+homepage = "https://github.com/rp-rs/rp-hal"
+description = "Examples for the rp2040-hal crate"
+license = "MIT OR Apache-2.0"
+rust-version = "1.77"
+repository = "https://github.com/rp-rs/rp-hal"
+categories = ["embedded", "hardware-support", "no-std", "no-std::no-alloc"]
+keywords = ["embedded", "hal", "raspberry-pi", "rp2040", "embedded-hal"]
+
+[dependencies]
+cortex-m = "0.7.2"
+cortex-m-rt = "0.7"
+cortex-m-rtic = "1.1.4"
+critical-section = { version = "1.0.0" }
+defmt = "0.3"
+defmt-rtt = "0.4.0"
+dht-sensor = "0.2.1"
+embedded_hal_0_2 = { package = "embedded-hal", version = "0.2.5", features = ["unproven"] }
+embedded-alloc = "0.5.1"
+embedded-hal = "1.0.0"
+embedded-hal-async = "1.0.0"
+fugit = "0.3.6"
+futures = { version = "0.3.30", default-features = false, features = ["async-await"] }
+hd44780-driver = "0.4.0"
+nb = "1.0"
+nostd_async = { version = "0.6.1", features = ["cortex_m"] }
+panic-halt = "0.2.0"
+panic-probe = { version = "0.3.1", features = ["print-defmt"] }
+pio = "0.2.0"
+pio-proc = "0.2.0"
+rp2040-boot2 = "0.3.0"
+rp2040-hal = { path = "../rp2040-hal", version = "0.10.0", features = ["binary-info", "critical-section-impl", "rt", "defmt"] }
```

Comment:
Although It was an eyesore, I think it was a good thing to see what features each example required.

Reply:
Feature unification. The HAL is only built once and needs all the features set that are required by any bin. With examples it would just silently not build any examples when the required feature wasn't set, which I found frustrating (`cargo build ` wouldn't build any examples, but `cargo build --all-features` would).

We could split each example into its own package, but that seems extreme.

Or I guess I could document each required feature at the top of each binary, but I've got no way of checking that that's correct.

Reply:
You are right. I found it nice that from a user experience, targeting a specific examples would tell "that examples requires those features".
But I’m fine with this as well.

Would it be good in the future to mutualize the examples and have a "portability"/"bsp" module in the example crate to map to either rp2040 or rp235x ?

Reply:
I think it's very unlikely that anyone *wouldn't* want the `critical-section-impl` or `rt` features. In particular, not using `rt` gives you a 'default' vector table which is broken and doesn't work. I'm also going to change all the examples so they include binary info by default, because it's just nice to have.

That leaves the 'defmt' feature. I could add a `defmt` feature to the examples package, and then mark the various binary crates within it that require the defmt feature in the HAL to require that feature to be set on the package.